docs/configuration-options.md

```diff
@@ -0,0 +1,115 @@
+
+## Table of Contents
+
+- [Step 1: Laying the Foundation – Establish Benchmark Targets](#step-1)
+- [Step 2: Tailoring the Setup – Create Benchmark Configurations](#step-2)
+- [Step 3: Fine-tuning Your Setup – Understanding and Setting Configuration Options](#step-3)
+  - [Basic Configuration Options: The Essential Settings](#step-3a)
+  - [Advanced Configuration Options: The Power Settings](#step-3b)
+
+## Step 1: Laying the Foundation – Establish Benchmark Targets <a name="step-1"></a>
+
+Your journey starts by defining the `benchmark` section within your `build.gradle` file. This section is your playground where you register the compilations you wish to run benchmarks on, within a `targets` subsection.
+
+Targets can be registered in two ways. Either by their name, such as `"jvm"`, which registers its `main` compilation, meaning `register("jvm")` and `register("jvmMain")` will register the same compilation. Alternatively, you can register a source set, for instance, `"jvmTest"` or `"jsBenchmark"`, which will register the corresponding compilation. Here's an illustration using a multiplatform project:
+
+```groovy
+benchmark {
+    targets {
+        register("jvm") 
+        register("js")
+        register("native")
+        register("wasm") // Experimental
+    }
+}
+```
+
+## Step 2: Tailoring the Setup – Create Benchmark Configurations <a name="step-2"></a>
+
+Having laid the groundwork with your targets, the next phase involves creating configurations for your benchmarks. You accomplish this by adding a `configurations` subsection within your `benchmark` block.
+
+The kotlinx benchmark toolkit automatically creates a `main` configuration as a default. However, you can mold this tool to suit your needs by creating additional configurations. These configurations are your control knobs, letting you adjust the parameters of your benchmark profiles. Here's how:
+
+```groovy
+benchmark {
+    configurations {
+        main { 
+            // Configuration parameters for the default profile go here
+        }
+        smoke { 
+            // Create and configure a "smoke" configuration.
+        }       
+    }
+}
+```
+
+## Step 3: Fine-tuning Your Setup – Understanding and Setting Configuration Options <a name="step-3"></a>
+
+Each configuration brings a bundle of options to the table, providing you with the flexibility to meet your specific benchmarking needs. We delve into these options to give you a better understanding and help you make the most of the basic and advanced settings:
+
+### Basic Configuration Options: The Essential Settings <a name="step-3a"></a>
+
+| Option | Description | Default Value | Possible Values |
+| --- | --- | --- | --- |
+| `iterations` | Specifies the number of iterations for measurements. | - | Integer |
```

Comment:
It is worth mentioning that many of the options correspond to annotations in benchmark code, and values provided in the build script override those values.

Reply:
This definitely seems necessary. I have updated the table to include this [6d43df4](https://github.com/Kotlin/kotlinx-benchmark/pull/117/commits/6d43df41dcdb8144dbe1412d1a7349a7473d507e)